stub-runner/stub-runner/src/main/groovy/com/ofg/stub/server/AvailablePortScanner.groovy

```diff
@@ -7,38 +7,50 @@ import groovy.util.logging.Slf4j
 @Slf4j
 class AvailablePortScanner {
 
+    public static final int MAX_RETRY_COUNT = 1000
     private final int minPortNumber
     private final int maxPortNumber
-    private int portToScan
 
     AvailablePortScanner(int minPortNumber, int maxPortNumber) {
-        this.portToScan = minPortNumber
         this.minPortNumber = minPortNumber
         this.maxPortNumber = maxPortNumber
     }
 
-    int nextAvailablePort() {
-        ServerSocket socket = null
-        while (portToScan <= maxPortNumber) {
+    void tryToExecuteWithFreePort(Closure closure) {
+        int counter = MAX_RETRY_COUNT
+        while (--counter > 0) {
             try {
-                socket = new ServerSocket(portToScan)
-                return portToScan
-            } catch (IOException e) {
-                log.debug("Port $portToScan not available")
-            } finally {
-                closeSocket(socket)
-                portToScan++
+                int portToScan = new Random().nextInt(maxPortNumber - minPortNumber) + minPortNumber
+                checkIfPortIsAvailable(portToScan)
```

Comment:
Technically you could skip `checkIfPortIsAvailable()` and just run closure. But your way avoid calling closure too many times, so +1.